pkg/util/httputil/deps.go

```diff
@@ -4,6 +4,7 @@ import "github.com/google/wire"
 
 var DependencySet = wire.NewSet(
 	NewJSONResponseWriterLogger,
+	NewHTTPClient,
```

Comment:
Wire inject by type. Adding NewHTTPClient will make it available in the graph, other components can then use it. But it is not our intention. So add a new type `type OAuthHTTPClient *http.Client` in `pkg/lib/authn/sso`. Add `NewOAuthHTTPClient` in `pkg/lib/authn/sso/deps.go`. Make OAuthProviderFactory to use OAuthHTTPClient instead of *http.Client.

Reply:
Think this will make `OAuthHTTPClient` unable to use methods of `*http.Client` 

If I use anonymous field though, I won't be able to intercept it with `gock.InterceptClient`.

Reply:
Updated to use custom struct for sso, thanks!